### PR TITLE
Fix 'block link explorer on custom networks'

### LINKED ISSUE #13262: Fix block link explorer on custom networks
### **User Story**
As a user, given I am on a custom network, when I click the block explorer icon in Activity details, I want to go to the correct block explorer for that network - today if the block explorer field is not added correctly, it shows the wrong explorer and gives an error.

**Figma/Notion links link:** 
https://www.notion.so/Fix-block-explorer-link-on-custom-networks-738376e799864b71b76793a588034dd2

### **Steps to reproduce**

1. Add a custom network on MetaMask, but don’t add the “Block Explorer URL” or add a wrong “Block Explorer URL”
2. Go to the initial screen on MetaMask
3. Click on the three dot menu next to your account
4. Notice the “View Account on Etherscan” copy or the wrong link
5. Click on it and notice MetaMask opening up an error page

###  Task/Acceptance criteria 

1. If there’s no “Block Explore URL” for the custom network, show the text “Add a block explorer” instead of “View on Etherscan/Block Explorer” or “View Asset in Explorer” for accounts, assets and transactions on the respective menus. @filipsekulic found five places when we have the option for update (on the Notion we have a three places): 
- On the three dot  accounts menu: 
image

- Account details - View on Etherscan
image
image

- On the three dot asset menu
image

- On the activity details:
image
image

- On the Send flow when a new account added
image
image

2. Clicking on "Add a block explorer" should take the user to the Edit Network page for the current network (screenshot below).

Screen Shot 2022-02-02 at 01 01 27

### Notes
Acceptance criteria updated based on the Barbara comment: https://github.com/MetaMask/metamask-extension/issues/13262#issuecomment-1027552173

### PR DESCRIPTION
**Pull request by filipsekulic**
Fixes: https://github.com/MetaMask/metamask-extension/issues/13262

Screencast:

https://user-images.githubusercontent.com/92310504/157636455-d71043d4-3097-4ad4-8bf9-49e2898d646f.mov

### CODE REVIEW

**Comment by mirjanaKukic** [comment]
Verified by QA

**Comment by adonesky1** [comment]
@filipsekulic looks like this needs a rebase

File: ui/components/app/menu-bar/account-options-menu.js [hunk 1]
```diff
@@ -67,9 +78,12 @@ export default function AccountOptionsMenu({ anchorElement, onClose }) {
...
       >
-        {rpcPrefs.blockExplorerUrl
-          ? t('viewinExplorer', [t('blockExplorerAccountAction')])
-          : t('viewOnEtherscan', [t('blockExplorerAccountAction')])}
+        {rpcPrefs.blockExplorerUrl &&
+          t('viewinExplorer', [t('blockExplorerAccountAction')])}
+        {!rpcPrefs.blockExplorerUrl && isCustomNetwork && t('addBlockExplorer')}
+        {!rpcPrefs.blockExplorerUrl &&
+          !isCustomNetwork &&
+          t('viewOnEtherscan', [t('blockExplorerAccountAction')])}
```
Comment (brad-decker) [inline ui/components/app/menu-bar/account-options-menu.js:86]: This is kind of hard to follow the logic, my suggestion would be to move this outside of the JSX return and instead do something like:

```js

const routeToAddBlockExplorerUrl = () => {
  history.push(`${NETWORKS_ROUTE}#blockExplorerUrl`);
};

const openBlockExplorer = () => {
  trackEvent({
    event: 'Clicked Block Explorer Link',
    category: 'Navigation',
    properties: {
      link_type: 'Account Tracker',
      action: 'Account Options',
      block_explorer_domain: getURLHostName(addressLink),
    },
  });
  global.platform.openTab({
    url: addressLink,
  });
  onClose();
};

// If the user is on a custom network and doesn't have a block explorer set in state the text
// will display "Add block explorer" and route the user to the network settings page to add
// a block explorer. If a block explorer url has been added and is known a new  tab will be
// opened to that block explorer and the text will say "view in explorer". Finally, the block
// explorer will not be set in the case of the MetaMask default networks, and the text will
// indicate that etherscan is the explorer and the link will open a new tab to etherscan.
let blockExplorerLinkText = t('addBlockExplorer');
let blockExplorerOnClick = routeToAddBlockExplorerUrl;
if (rpcPrefs.blockExplorerUrl) {
  blockExplorerLinkText = t('viewInExplorer', [t('blockExplorerAccountAction')]);
  blockExplorerOnClick = openBlockExplorer;
} else if (isCustomNetwork === false) {
  blockExplorerLinktext = t('viewOnEtherscan', [t('blockExplorerAccountAction')]);
  blockExplorerOnClick = openBlockExplorer;
}
```

disclaimer: This is pseudo-pseudo code written in browser, please review it, apply the *concept* to your work if you agree. :)

Comment (brad-decker) [reply ui/components/app/menu-bar/account-options-menu.js:86]: PS: IT seems that this logic is done in many places. It might be beneficial to change this into a redux selector, action or a react hook (please use your best judgement) to prevent duplication of this logic as much as possible.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/app/menu-bar/account-options-menu.js (hunk 1 above)
Comment (filipsekulic) [reply ui/components/app/menu-bar/account-options-menu.js]: @brad-decker 
I did what you had suggested using the redux selector in this case. The new logic can be seen here: a40ddd64347037c7fc7ddee838263712cd149c92
Hope I got the most out of it. :) 
Also, these are the options that were considered during the code refactoring:

- addBlockExplorer
- viewinExplorer
- viewOnEtherscan 
- viewOnBlockExplorer
- blockExplorerView
- etherscanViewOn
- etherscanView

**Review by brad-decker — APPROVED** [review]
LGTM, sorry for the delay in reviewing this.

**Comment by filipsekulic** [comment]
> LGTM, sorry for the delay in reviewing this.

No problem. Thank you for reviewing it again!

**Review by adonesky1 — APPROVED** [review]
